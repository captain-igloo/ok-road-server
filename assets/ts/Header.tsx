import * as React from 'react';

import UserMenu from './UserMenu';

interface Props {
    showDemo?: boolean;
    showMap?: boolean;
    showRegister?: boolean;
    showSignIn?: boolean;
    user?: {
        fullName: string;
    };
}

export default function Header(props: Props) {
    const {
        showDemo,
        showMap,
        showRegister,
        showSignIn,
        user,
    } = props;

    return (
        <div className="header">
            <div className="logo">
                <a href="/">
                    <img alt="Home" src="/img/ok.svg" width="40px" height="40px" />
                </a>
                &nbsp;
                OK Road New Zealand
            </div>
            <UserMenu
                showDemo={showDemo}
                showMap={showMap}
                showRegister={showRegister}
                showSignIn={showSignIn}
                user={user}
            />
        </div>
    );
}
