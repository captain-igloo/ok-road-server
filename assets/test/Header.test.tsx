import { describe, expect, test } from '@jest/globals';
import { render } from '@testing-library/react';
import * as React from 'react';

import Header from '../ts/Header';

describe('Header component', () => {
    test('Header renders properly', () => {
        const { container } = render(<Header showDemo showMap showRegister showSignIn />);
        expect(container).toMatchSnapshot();
    });
});
