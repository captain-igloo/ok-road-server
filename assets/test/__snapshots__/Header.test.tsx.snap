// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Header renders properly 1`] = `
<div>
  <div
    class="header"
  >
    <div
      class="logo"
    >
      <a
        href="/"
      >
        <img
          alt="Home"
          height="40px"
          src="/img/ok.svg"
          width="40px"
        />
      </a>
        OK Road New Zealand
    </div>
    <div
      class="btn-group"
      role="group"
    >
      <a
        class="btn btn-outline"
        href="/demo"
        role="button"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-person-chalkboard "
          data-icon="person-chalkboard"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M192 96a48 48 0 1 0 0-96 48 48 0 1 0 0 96zm-8 384l0-128 16 0 0 128c0 17.7 14.3 32 32 32s32-14.3 32-32l0-288 56 0 64 0 16 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-16 0 0-64 192 0 0 192-192 0 0-32-64 0 0 48c0 26.5 21.5 48 48 48l224 0c26.5 0 48-21.5 48-48l0-224c0-26.5-21.5-48-48-48L368 0c-26.5 0-48 21.5-48 48l0 80-76.9 0-65.9 0c-33.7 0-64.9 17.7-82.3 46.6l-58.3 97c-9.1 15.1-4.2 34.8 10.9 43.9s34.8 4.2 43.9-10.9L120 256.9 120 480c0 17.7 14.3 32 32 32s32-14.3 32-32z"
            fill="currentColor"
          />
        </svg>
        <span
          class="display-small"
        >
            Demo
        </span>
      </a>
      <a
        class="btn btn-outline"
        href="/register"
        role="button"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-user-plus "
          data-icon="user-plus"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 128a128 128 0 1 1 256 0A128 128 0 1 1 96 128zM0 482.3C0 383.8 79.8 304 178.3 304l91.4 0C368.2 304 448 383.8 448 482.3c0 16.4-13.3 29.7-29.7 29.7L29.7 512C13.3 512 0 498.7 0 482.3zM504 312l0-64-64 0c-13.3 0-24-10.7-24-24s10.7-24 24-24l64 0 0-64c0-13.3 10.7-24 24-24s24 10.7 24 24l0 64 64 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-64 0 0 64c0 13.3-10.7 24-24 24s-24-10.7-24-24z"
            fill="currentColor"
          />
        </svg>
        <span
          class="display-small"
        >
            Register
        </span>
      </a>
      <a
        class="btn btn-outline"
        href="/login"
        role="button"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-right-to-bracket "
          data-icon="right-to-bracket"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M217.9 105.9L340.7 228.7c7.2 7.2 11.3 17.1 11.3 27.3s-4.1 20.1-11.3 27.3L217.9 406.1c-6.4 6.4-15 9.9-24 9.9c-18.7 0-33.9-15.2-33.9-33.9l0-62.1L32 320c-17.7 0-32-14.3-32-32l0-64c0-17.7 14.3-32 32-32l128 0 0-62.1c0-18.7 15.2-33.9 33.9-33.9c9 0 17.6 3.6 24 9.9zM352 416l64 0c17.7 0 32-14.3 32-32l0-256c0-17.7-14.3-32-32-32l-64 0c-17.7 0-32-14.3-32-32s14.3-32 32-32l64 0c53 0 96 43 96 96l0 256c0 53-43 96-96 96l-64 0c-17.7 0-32-14.3-32-32s14.3-32 32-32z"
            fill="currentColor"
          />
        </svg>
        <span
          class="display-small"
        >
            Sign in
        </span>
      </a>
    </div>
  </div>
</div>
`;
